### PR TITLE
Have a question#20

### PR DESCRIPTION
**Pull request by baraahmad**

### CODE REVIEW

**Review by ddfridley — CHANGES_REQUESTED** [review]
This is great.  I added nodemailerstart to the dependencies in package.json so now npm run dev works, and sending to the server works.  You should be able to do a git pull to get this change.  then you will need to do an npm install.

I want to request one improvement though.  After success, or a failure, there is a blue box at the bottom with a message.  

- [x] After 10 seconds, make the blue box at the bottom go away.
- [x] After 10 seconds, If after sending there there was a failure, reset the state so the user can enter their email and hit submit again.  

Thanks!

File: app/components/have-a-question.js [hunk 1]
```diff
@@ -100,41 +118,88 @@ const useStyles = createUseStyles(theme => ({
...
     resize: 'none',
     border: 'none',
     borderRadius: theme.defaultBorderRadius,
     marginBottom: '0.3rem',
     marginTop: '0.3rem',
     overflow: 'hidden',
+    fontWeight: 500,
+    padding: '0.9375rem 0rem',
+    borderTop: 'none',
+    borderBottom: '2px solid',
```
Comment (ddfridley) [inline app/components/have-a-question.js:134]: Even for border, please calculate the rems to use here.

File: app/components/have-a-question.js [hunk 2]
```diff
@@ -47,46 +49,77 @@ export default function HaveAQuestion(props) {
...
         let { error } = response
         setResponse(error)
+        setErrorMessage(true)
       } else {
         setResponse('Your question was sucessfully submitted!')
       }
+      setIsAlertVisible(true)
+
+      setTimeout(() => {
+        setIsAlertVisible(false)
```
Comment (ddfridley) [inline app/components/have-a-question.js:59]: When the operation is successful it works great.  But when "somethings worng" (you don't type in success@email.com) you are suck not being able to try again.

There can be a different setTimeout for the case when there is an error vs when there isn't.

When there is an error, after the 10 seconds, I think we should leave the email address and submit button visible but clear the error message, and allow the user to change the email address if necessary and hit submit again.

Managing the states, and making it easy to understand is a trick thing, but a good thing to get good at. Lets try this a little more.

**Review by ddfridley — CHANGES_REQUESTED** [review]
This looks great, and the happy case works great. But when there's an error, I'd like to do a little better and give the user the ability to try again.  See the comment.

Thanks!

File: app/components/have-a-question.js (hunk 2 above)
Comment (baraahmad) [reply app/components/have-a-question.js:59]: Sure!

**Review by ddfridley — APPROVED** [review]
You did it1 It looks great. Thanks!

**Pull request merged**